plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -218,7 +218,7 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []v1.Volume, namespace s
 				return fmt.Errorf("PersistentVolumeClaim had no name")
 			}
 			pvc, err := c.pvcInfo.GetPersistentVolumeClaimInfo(namespace, pvcName)
-			if err != nil {
+			if err != nil || pvc == nil {
```

Comment:
I don't understand this. Why should we return nil (no error) when the PVC is not found?

cc/ @msau42

Reply:
This predicate is counting the number of PVs that are running on a single Node across all namespaces and Pods.

Users can delete PVC objects even when a Pod is still running and using it.  So even though the PVC object doesn't exist anymore, the volume could still be mounted and used on the node, so we should count it since the Pod is referencing it.